src/fiber/tests/payment.rs

```diff
@@ -940,58 +741,506 @@ async fn test_send_payment_two_nodes_with_hop_hints_and_multiple_channels() {
             max_parts: None,
             keysend: Some(true),
             udt_type_script: None,
-            allow_self_payment: true,
+            allow_self_payment: false,
             hop_hints: Some(vec![
-                // node1 - channel_1 -> node2
                 HopHint {
-                    pubkey: node_0.pubkey.clone(),
-                    channel_funding_tx: channel_1_funding_tx,
-                    inbound: false,
+                    pubkey: node2.pubkey.clone(),
+                    channel_outpoint: outpoint1,
+                    fee_rate: DEFAULT_TLC_FEE_PROPORTIONAL_MILLIONTHS as u64,
+                    tlc_expiry_delta: DEFAULT_TLC_EXPIRY_DELTA,
                 },
-                // node2 - channel_3 -> node1
                 HopHint {
-                    pubkey: node_0.pubkey.clone(),
-                    channel_funding_tx: channel_3_funding_tx,
-                    inbound: true,
+                    pubkey: node2.pubkey.clone(),
+                    channel_outpoint: outpoint2,
+                    fee_rate: DEFAULT_TLC_FEE_PROPORTIONAL_MILLIONTHS as u64,
+                    tlc_expiry_delta: DEFAULT_TLC_EXPIRY_DELTA,
                 },
             ]),
             dry_run: false,
         })
-        .await
-        .unwrap();
+        .await;
 
+    assert!(res.is_ok(), "Send payment failed: {:?}", res);
+    let res = res.unwrap();
     let payment_hash = res.payment_hash;
-    eprintln!("payment_hash: {:?}", payment_hash);
-    let payment_session = node_0
-        .get_payment_session(payment_hash)
-        .expect("get payment");
-    eprintln!("payment_session: {:?}", payment_session);
-    let used_channels: Vec<Hash256> = payment_session
-        .route
-        .nodes
-        .iter()
-        .map(|x| x.channel_outpoint.tx_hash().into())
-        .collect();
-    eprintln!("used_channels: {:?}", used_channels);
-    assert_eq!(used_channels.len(), 3);
-    assert_eq!(used_channels[0], channel_1_funding_tx);
-    assert_eq!(used_channels[1], channel_3_funding_tx);
-
-    tokio::time::sleep(tokio::time::Duration::from_millis(2000)).await;
-
-    let balance = node_0.get_local_balance_from_channel(channels[1]);
-    assert_eq!(balance, old_balance - 60000000 - res.fee);
-
-    let node_1_balance = node_1.get_local_balance_from_channel(channels[1]);
-    assert_eq!(node_1_balance, 60000000 + res.fee);
-
-    let balance = node_0.get_local_balance_from_channel(channels[3]);
-    assert_eq!(balance, 60000000);
-
-    let node_1_balance = node_1.get_local_balance_from_channel(channels[3]);
-    assert_eq!(node_1_balance, old_node1_balance - 60000000);
+    source_node.wait_until_success(payment_hash).await;
 }
 
+// #[tokio::test]
+// async fn test_send_payment_with_route_to_self_with_hop_hints() {
+//     init_tracing();
```

Comment:
we expect to add these tests back after we have `send_to_route`?

Reply:
Yes.